src/main/java/org/candlepin/database/DatabaseConnectionManager.java

```diff
@@ -0,0 +1,144 @@
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Objects;
+
+/**
+ * Class for creating and maintaining a connection to the database.
+ */
+public class DatabaseConnectionManager {
+    private static Logger log = LoggerFactory.getLogger(DatabaseConnectionManager.class);
+
+    private Configuration config;
+    private Connection connection;
+
+    /**
+     * Creates a database connection based on the defined configuration values.
+     *
+     * @throws LiquibaseException
+     *  if unable to establish a connection to the database
+     *
+     * @throws ReflectiveOperationException
+     *  if unable to load the database driver
+     *
+     * @param config
+     *  used to retrieve the database configuration values needed to establish a connection
+     */
+    public DatabaseConnectionManager(Configuration config)
+        throws LiquibaseException, ReflectiveOperationException {
+        this.config = Objects.requireNonNull(config);
+
+        loadDatabaseDriver();
+        this.connection = createConnection();
+    }
+
+    /**
+     * @return the established database connection
+     */
+    public Connection getConnection() {
+        return this.connection;
+    }
+
+    /**
+     * Creates a {@link Database} object based on the database configuration.
+     *
+     * @throws LiquibaseException
+     *  if unable to create the {@link Database} from the database connection
+     *
+     * @return the database created from the database connection
+     */
+    public Database getDatabase() throws LiquibaseException {
+        JdbcConnection jdbcConnection = new JdbcConnection(this.connection);
+
+        return DatabaseFactory.getInstance()
+            .findCorrectDatabaseImplementation(jdbcConnection);
+    }
+
+    /**
+     * Creates a database connection based on database configuration values. If a connection is not initially
+     * created, further attempts will be made to create a connection. If a connection cannot be created after
+     * the configured max attempts, then a {@link LiquibaseException} will be thrown.
+     *
+     * @throws LiquibaseException
+     *  if unable to create a database connection after max retry attempts
+     *
+     * @return
+     *  the database connection
+     */
+    private Connection createConnection() throws LiquibaseException {
+        int attemptNumber = 1;
+        int maxAttempts = config.getInt(ConfigProperties.DB_MAX_CONNECTION_ATTEMPTS);
+
+        while (attemptNumber <= maxAttempts) {
+            log.info("Attempt {} out of {} to connect to the database.", attemptNumber, maxAttempts);
+            try {
+                return DriverManager.getConnection(
+                    config.getString(ConfigProperties.DB_URL),
+                    config.getString(ConfigProperties.DB_USERNAME),
+                    config.getString(ConfigProperties.DB_PASSWORD));
+            }
+            catch (SQLException e) {
```

Comment:
Is this always going to be a SQLException when we can't connect to the DB? Or is there any circumstance it can be something else, like IOException (with e.g. some kind of network issues)?

Reply:
I think it will always be SQLException, but we can just catch Exception here to be sure! Updated